### PR TITLE
ARROW-9252: [Integration] Factor out IPC integration tests into script, add back 0.14.1 "gold" files

### PR DESCRIPTION
**Pull request by wesm**

### CODE REVIEW

**Comment by wesm** [comment]
Looks like the int64 tests must be removed from the "gold" corpus as the JSON files cannot be parsed anymore

**Comment by wesm** [comment]
@emkornfield looks like almost none of the 0.14.1 "gold" files can be read anymore because of the change to represent int64 values as strings. We should prepare a set of gold files from after the int64 changes but before any patches that alter the MetadataVersion etc. Do you have a script that you used to create the gold files?

**Comment by emkornfield** [comment]
No, IIRC I think I hacked integration tests to generate smaller files then looped over them with one of the binaries.  I can try to come up with something to update the json files and submit a PR to the test data repo?

**Comment by wesm** [comment]
Sure, that sounds good

File: ci/scripts/integration_arrow.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# under the License.
+
+set -ex
+
+arrow_dir=${1}
+source_dir=${1}/cpp
+build_dir=${2}/cpp
+gold_dir_0_14_1=${1}/testing/data/arrow-ipc-stream/integration/0.14.1
+
+pip install -e /arrow/dev/archery
```
Comment (pitrou) [inline ci/scripts/integration_arrow.sh:27]: Use `$source_dir` here?

**Comment by wesm** [comment]
Will merge this if the build passes with the arrow-testing changes

**Comment by wesm** [comment]
Yahtzee

**Pull request closed without merging**